### PR TITLE
[SRP] 2021-06-01 Swagger Api

### PR DESCRIPTION
**Pull request by HimanshuChhabra**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/examples/StorageAccountCreateWithImmutabilityPolicy.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/examples/StorageAccountCreateWithImmutabilityPolicy.json:1]: As VLW is also available in update account , please also add update account sample.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json:996]: Please also update example

Comment (HimanshuChhabra) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json:996]: It is part of existing published GA swagger

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:2604]: What's "none" means? Can customer set "None" from rest API?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json]: what's the difference of set this property to true and false?
IS there any feature spec?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/examples/StorageAccountCreateWithImmutabilityPolicy.json
Comment (HimanshuChhabra) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/examples/StorageAccountCreateWithImmutabilityPolicy.json:1]: done

**Comment by mentat9** [comment]
@Himanshg, please create a new PR following this item (and check the box when you submit): **[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.** This is very important to get an quality and efficient ARM review.

**Comment by HimanshuChhabra** [comment]
> @Himanshg, please create a new PR following this item (and check the box when you submit): **[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.** This is very important to get an quality and efficient ARM review.

>@mentat9, this is exactly how this PR has been prepared. Below the very first commit which is the directory copy from older version to the new version.
https://github.com/Azure/azure-rest-api-specs/pull/15627/commits/7e0f2751aed1705d5a1857979c629198f645e3fb

All the new commits thereafter are additional changes only for the new version as separate commts

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json:1003]: Can "enableNfsV3RootSquash" and "enableNfsV3AllSquash" both be enabled?  Is there any limitation between these 2 properties, and between the storage account property "EnableNfsV3"? 
Ask this since in File share, can only choose one from RootSquash/AllSquash.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json:998]: It seems the feature still not work in my subscription in canary.
Do I need to enable any feature flag to run the feature?

```
PUT https://management.azure.com/subscriptions/{subcriptionID}/resourceGroups/res6672/providers/Microsoft.Storage/storageAccounts/sto2847/blobServices/default/containers/container2042?api-version=2021-06-01 HTTP/1.1
x-ms-client-request-id: 232013b6-fc21-4eac-8b5a-3072da1c5280
Accept-Language: en-US
Authorization: Bearer ***
User-Agent: FxVersion/4.6.30321.06 OSName/Windows OSVersion/Microsoft.Windows.10.0.19043. Microsoft.Azure.Management.Storage.StorageManagementClient/22.0.0.0
Content-Type: application/json; charset=utf-8
Content-Length: 63
Host: management.azure.com

{
  "properties": {
    "enableNfsV3RootSquash": true
  }
}
HTTP/1.1 400 Bad Request
Cache-Control: no-cache
Pragma: no-cache
Content-Length: 118
Content-Type: application/json
Expires: -1
x-ms-request-id: 610f3434-d8fe-47cc-84c0-e56f7558d6d4
Strict-Transport-Security: max-age=31536000; includeSubDomains
Server: Microsoft-Azure-Storage-Resource-Provider/1.0,Microsoft-HTTPAPI/2.0 Microsoft-HTTPAPI/2.0
x-ms-ratelimit-remaining-subscription-writes: 1197
x-ms-correlation-request-id: 01f680f4-ea12-4454-bd69-7b85e4fe4897
x-ms-routing-request-id: SOUTHEASTASIA:20210816T094055Z:01f680f4-ea12-4454-bd69-7b85e4fe4897
X-Content-Type-Options: nosniff
Date: Mon, 16 Aug 2021 09:40:54 GMT

{"error":{"code":"FeatureNotSupportedForAccount","message":"EnableNfsV3RootSquash is not supported for the account."}}
```

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:3601]: Update account level VLW in my test fail, repro step:
1. create account with account level VLW enabled
2. update account to disable it, Get error `{"error":{"code":"PropertyIsImmutable","message":"The property 'immutableStorageWithVersioning.enabled' was specified in the input, but it cannot be updated as it is immutable."}}`, request ID `x-ms-request-id: c8f2df85-9cf8-4561-b48b-fa5f724a49e3`

Is the error expected? Should we update property description to include the limitation?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json:1001]: Please add example for enableNfsV3RootSquash and enableNfsV3AllSquash

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:139]: per my test, create account with account level WORM will fail with error `{"error":{"code":"InvalidStorageAccountImmutabilityPolicy","message":"Storage account level default immutability policy cannot be set as account level WORM is disabled"}}`, request ID `x-ms-request-id: b5293f9d-3fab-461a-9c9e-fa2b8b5901c8`

So if we should only set true to the "enabled" value (since update with "enabled" as false also fail), we might should not set this property in create/update account.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (HimanshuChhabra) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:3601]: The description already says that the property is immutable and the server behavior is expected

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (HimanshuChhabra) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:139]: This is by compliance. Customer has to explicitly enable Account level VLW - ImmutableStorageWithVersioning.enabled = true
Only then ImmutableStorageWithVersioning.immutabilityPolicy can be set. 
This is expected server behavior

ImmutableStorageWithVersioning.enabled is immutable and is mentioned in the description of the model

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/examples/StorageAccountAbortHnsOnMigration.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/examples/StorageAccountAbortHnsOnMigration.json]: Abort HNSon Migration always fail as 400 in my test, see network trace below.
Step:
1. invoke migration with type "HnsOnHydrationRequest"
2. wait for 10 s
3. invoke about migration request , fail with 400

```
POST https://management.azure.com/subscriptions/{SubId}/resourcegroups/res4933/providers/Microsoft.Storage/storageAccounts/sto7451/hnsonmigration?api-version=2021-06-01&requestType=HnsOnHydrationRequest HTTP/1.1
x-ms-client-request-id: 22d0615f-92b2-4450-9ec3-ee9b4afa2b11
Accept-Language: en-US
Authorization: Bearer ***
User-Agent: FxVersion/4.6.30321.06 OSName/Windows OSVersion/Microsoft.Windows.10.0.19043. Microsoft.Azure.Management.Storage.StorageManagementClient/22.0.0.0
Content-Length: 0
Host: management.azure.com


HTTP/1.1 202 Accepted
Cache-Control: no-cache
Pragma: no-cache
Content-Type: text/plain; charset=utf-8
Expires: -1
Location: https://management.azure.com/subscriptions/45b60d85-fd72-427a-a708-f994d26e593e/providers/Microsoft.Storage/locations/centraluseuap/asyncoperations/7001fe91-53d0-4aaf-a045-edfa9cb81583?monitor=true&api-version=2021-06-01
Retry-After: 17
Azure-AsyncOperation: https://management.azure.com/subscriptions/45b60d85-fd72-427a-a708-f994d26e593e/providers/Microsoft.Storage/locations/centraluseuap/asyncoperations/7001fe91-53d0-4aaf-a045-edfa9cb81583?monitor=true&api-version=2021-06-01
x-ms-request-id: 7001fe91-53d0-4aaf-a045-edfa9cb81583
Strict-Transport-Security: max-age=31536000; includeSubDomains
Server: Microsoft-Azure-Storage-Resource-Provider/1.0,Microsoft-HTTPAPI/2.0 Microsoft-HTTPAPI/2.0
x-ms-ratelimit-remaining-subscription-writes: 1195
x-ms-correlation-request-id: 60df778c-8b05-4096-8515-dad1150d84e6
x-ms-routing-request-id: SOUTHEASTASIA:20210816T100136Z:60df778c-8b05-4096-8515-dad1150d84e6
X-Content-Type-Options: nosniff
Date: Mon, 16 Aug 2021 10:01:36 GMT
Content-Length: 0



------------------------------------------------------------------
POST https://management.azure.com/subscriptions/{SubId}/resourcegroups/res4933/providers/Microsoft.Storage/storageAccounts/sto7451/aborthnsonmigration?api-version=2021-06-01 HTTP/1.1
x-ms-client-request-id: 7b59fb82-ea06-4787-839d-f5083e9f9d59
Accept-Language: en-US
Authorization: Bearer ***
User-Agent: FxVersion/4.6.30321.06 OSName/Windows OSVersion/Microsoft.Windows.10.0.19043. Microsoft.Azure.Management.Storage.StorageManagementClient/22.0.0.0
Content-Length: 0
Host: management.azure.com


HTTP/1.1 400 Bad Request
Cache-Control: no-cache
Pragma: no-cache
Content-Length: 106
Content-Type: application/json
Expires: -1
x-ms-request-id: 4fcd9f35-0f1d-4ec9-988e-35b290485c22
Strict-Transport-Security: max-age=31536000; includeSubDomains
Server: Microsoft-Azure-Storage-Resource-Provider/1.0,Microsoft-HTTPAPI/2.0 Microsoft-HTTPAPI/2.0
x-ms-ratelimit-remaining-subscription-writes: 1194
x-ms-correlation-request-id: a46cade3-e62f-469f-b877-b34b3f8c3aa0
x-ms-routing-request-id: SOUTHEASTASIA:20210816T100146Z:a46cade3-e62f-469f-b877-b34b3f8c3aa0
X-Content-Type-Options: nosniff
Date: Mon, 16 Aug 2021 10:01:46 GMT

{"error":{"code":"HnsMigrationNotFound","message":"Hns migration for the account: sto7451 is not found."}}

------------------------------------------------------------------
```

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:139]: This is not a good design:
If it can only be set to true, we might should not use bool as the type, to avoid unnecessary invalid user input. 
Or might should not allow user to input it, as it's single value. When user input an immutability policy, the value should set to true automatically. 
Besides that, the limitation for only true is allowed value, should be mentions in description.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:3601]: If user can't change it in update account, why add it to update account parameters? 
All properties here should be allowed to update in update storage account API.

Comment (HimanshuChhabra) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:3601]: ImmutableStorageWithVersioning is an object only 1 property enabled is immutable, the ImmutableStorageWithVersioning.immutablityPolicy is mutable

Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:3601]: The description of property "immutableStorageWithVersioning" says "The property is immutable...", so for common understanding, the whole property is immutable. The description need change to make it more clear.

Besides that, as the value of "enabled" should always be true, can't be other value in create/update account, we might should not add it to rest API, to avoid user error.

Comment (HimanshuChhabra) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:3601]: I will update the comment, However, it is absolutely legal to set enabled=false to account creation time.
Also, we need enabled property here because immutablityPolicy is optional the enabled property will enable VLW for all the containers under such an account without any account level immutabilityPolicy. 
The enabled = true will allow customers to exercise blob level immutability apis.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (HimanshuChhabra) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:139]: There is a gap here, immutabilityPolicy is an OPTIONAL FIeld. So we cannot get rid of enabled property

It is absolutely legal to set enabled=false to account creation time. 
By setting enabled=true , all the containers created under such an account will be VLW enabled and user can set ImmutabilityPolicy at the blob level, or optionally at container level using CLW apis or account level using this model.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (raomanus) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:2604]: Below is the expected behavior for the above property.
1. Enabled & None = Allow client connection requests.
2. Disabled = Block connections over public endpoints even if they match the IP/VNet firewall rules.

Will share the sample rest APIs for this offline.

Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:2604]: Since "None is same as "Enabled", Why we need "None"? Is it the default value?
Can customer set "None" from rest API?  Or none is readonly?

From many other properties only have enabled/disable status, we use a bool value type instead of enum.
The .net SDK will have it as "bool?". The default value is server not return it, so .net SDK will have it as null. 
You can reference https://github.com/Azure/azure-rest-api-specs/blob/3893616381e816729ef9cdd768e87fb2845e189d/specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/storage.json#L3044

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:139]: >  immutabilityPolicy is an OPTIONAL FIeld. So we cannot get rid of enabled property
[Wei] Actually, you don't need enable property:  In create account, just don't add the worm policy in create account request, will mean the account don't have account level worm policy. In update account, actually, the value should always be "true" when set worm policy, so add the enable property don't add any mire info to the update account rest request.

> It is absolutely legal to set enabled=false to account creation time.
[Wei] Server doesn't allow it, Set enabled=false to account creation , will get error `{"error":{"code":"InvalidStorageAccountImmutabilityPolicy","message":"Storage account level default immutability policy cannot be set as account level WORM is disabled"}}`

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:2552]: Per feature team, "StorageFileDataSmbShareOwner" an unsupported value in SRP , We have laready remove it from PSH/CLI. Should we also remove it from swagger?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (raomanus) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:2604]: 1. Yes, "None" is equal to the default value (null in case the property is never set)
2. Customer can set the property to "None" from rest API. It is not readonly.
3. The requirement for the property stated that the values should be strings and not bool which is why we have used the string values. In the future we may allow additional values for this property and in those cases a bool would not be sufficient.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (HimanshuChhabra) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:139]: Customers can simply enable account level WORM in CREATE and want to add Immutability Policy in UPDATE. We cannot have two different models for CREATE and UPDATE flows. 
enabled being immutable and immuatbilityPolicy being mutable

Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:139]: Actually, per my test, in both create/update storage account, "enabled" must be true or server will report error.
If this is expect, "Enabled" is not a necessary parameter, as it must be true
If this is not expected, please check why create/update account  with "enabled" as false fail on server

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:3601]: > it is absolutely legal to set enabled=false to account creation time.

In my test, create account with set enabled=false with error {"error":{"code":"InvalidStorageAccountImmutabilityPolicy","message":"Storage account level default immutability policy cannot be set as account level WORM is disabled"}}, request ID x-ms-request-id: b5293f9d-3fab-461a-9c9e-fa2b8b5901c8

**Comment by jorgecotillo** [comment]
> @Himanshg Himanshu Gupta FTE, please create a new PR following this item (and check the box when you submit): **[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.** This is very important to get an quality and efficient ARM review.

I don't see this message being ack'd, we cannot proceed with the API review until you follow these steps.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:2604]: @raomanus 
Please indicate how comes "The requirement for the property stated that the values should be strings and not bool "
It still looks odd that has 2 option Enabled & None with same meaning.

Besides that, please answer: can we set "None" when create/update account? If so, is there any difference than set "enabled" in create/update account?

**Review by blueww — APPROVED** [review]
Please remove  unsupported "StorageFileDataSmbShareOwner" before merge.

**Comment by jianyexi** [comment]
There is a cross version breaking change , please follow the action https://github.com/Azure/azure-rest-api-specs/pull/15627#issuecomment-906986199  to get approval
image

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json:790]: We might should remove the "default" responds handling part, since:
1. Without  the default part, the error will be handled by default CloudException in .net SDK, and the error details will be in exception message. 
2. With the default part, the error will be handle by storage defined "#/definitions/ErrorResponse", the returned exception in .net SDK will only has error code in exception message, but won't has detail error message in the exception message. It will be difficult for customer to find the detail error message, as they need go into the exception properties and  try to find which property contains the error.

As the default error handling can handle error with following body very well. I would suggest:

-  if the error body of an API have following format, we should not add "default" responds part for it.
- only add default responds handling for error , when the error body is not in following format.

```
Body:
{
  "error": {
    "code": "***",
    "message": "****"
  }
}
```

For you reference, following are errors from .net SDK:
Error from API without default responds section:
```
Microsoft.Rest.Azure.CloudException: An operation is currently performing on this storage account that requires exclusive access.
   at Microsoft.Azure.Management.Storage.StorageAccountsOperations.<DeleteWithHttpMessagesAsync>d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.Storage.StorageAccountsOperationsExtensions.<DeleteAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.Storage.StorageAccountsOperationsExtensions.Delete(IStorageAccountsOperations operations, String resourceGroupName, String accountName)
   at Microsoft.Azure.Commands.Management.Storage.RemoveAzureStorageAccountCommand.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
```
Error from API with default responds section:
```
Microsoft.Azure.Management.Storage.Models.ErrorResponseException: Operation returned an invalid status code 'BadRequest'
   at Microsoft.Azure.Management.Storage.StorageAccountsOperations.<BeginHierarchicalNamespaceMigrationWithHttpMessagesAsync>d__23.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Azure.Management.Storage.StorageAccountsOperations.<HierarchicalNamespaceMigrationWithHttpMessagesAsync>d__17.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Azure.Management.Storage.StorageAccountsOperationsExtensions.<HierarchicalNamespaceMigrationAsync>d__25.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.Azure.Management.Storage.StorageAccountsOperationsExtensions.HierarchicalNamespaceMigration(IStorageAccountsOperations operations, String resourceGroupName, String accountName, String requestType)
   at Microsoft.Azure.Commands.Management.Storage.InvokeAzureStorageAccountHierarchicalNamespaceUpgradeCommand.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
```

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json:1058]: (Optional ) We should change the ` "name": "parameters",` to  "name": "immutabilityPolicy", on line 532 and 750 in blob.json. 

If not change, the generated SDK will has the parameter name for input immutabilityPolicy as "parameters", which doesn't make sense.
Before add "allowProtectedAppendWritesAll", this issue not appear in SDK, since immutabilityPolicy  only has 2 properties can be input, the Set/extend immutabilityPolicy  API will have the 2 immutabilityPolicy  properties just as the API paramters of the API; but after add one more property "allowProtectedAppendWritesAll", the Set/extend immutabilityPolicy  API will have the whole immutabilityPolicy  has parameter. 

The .net SDK API with current swagger (See "ImmutabilityPolicy parameters = default(ImmutabilityPolicy) ")

```
public static ImmutabilityPolicy CreateOrUpdateImmutabilityPolicy(this IBlobContainersOperations operations, string resourceGroupName, string accountName, string containerName, ImmutabilityPolicy parameters = default(ImmutabilityPolicy), string ifMatch = default(string))
```

The API after change  ` "name": "parameters",` to  "name": "immutabilityPolicy" in swagger: (See "ImmutabilityPolicy immutabilityPolicy = default(ImmutabilityPolicy)")
```
public static ImmutabilityPolicy CreateOrUpdateImmutabilityPolicy(this IBlobContainersOperations operations, string resourceGroupName, string accountName, string containerName, ImmutabilityPolicy immutabilityPolicy = default(ImmutabilityPolicy), string ifMatch = default(string))
```

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json:328]: WE should also add "AllowProtectedAppendWritesAll" to account level worm.

Comment (HimanshuChhabra) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/blob.json:328]: Not supported right now

**Pull request merged**
